tests/testSphPoint.py

```diff
@@ -0,0 +1,615 @@
+        self.assertTrue(point1 == point1)
+        self.assertFalse(point1 != point1)
+
+        lon, lat = self._dataset[1]
+        point2 = SphPoint(lon*degrees, lat*degrees)
+        self.assertTrue(point2 != point1)
+        self.assertTrue(point1 != point2)
+        self.assertFalse(point2 == point1)
+        self.assertFalse(point1 == point2)
+
+        pointCopy = copy.deepcopy(point1)
+        self.assertEqual(pointCopy, point1)
+        self.assertEqual(point1, pointCopy)
+        self.assertFalse(pointCopy != point1)
+        self.assertFalse(point1 != pointCopy)
+
+        # Test for transitivity (may be assumed by algorithms).
+        for delta in [10.0**(0.1*x) for x in range(-150, -49, 5)]:
+            self._checkTransitive(delta)
+
+    def _checkTransitive(self, delta):
+        """Test if equality is transitive even for close points.
+
+        This test prevents misuse of approximate floating-point
+        equality -- if `__eq__` is implemented using AFP, then this
+        test will fail for some value of `delta`. Testing multiple
+        values is recommended.
+
+        Parameters
+        ----------
+        delta : `number`
+            The separation, in degrees, at which point equality may
+            become intransitive.
+        """
+        point1 = SphPoint((45.0 - delta)*degrees, -18.0*degrees);
+        point2 = SphPoint( 45.0         *degrees, -18.0*degrees);
+        point3 = SphPoint((45.0 + delta)*degrees, -18.0*degrees);
+
+        self.assertTrue(point1 != point2 or point2 != point3 or point1 == point3)
+        self.assertTrue(point3 != point1 or point1 != point2 or point3 == point2)
+        self.assertTrue(point2 == point3 or point3 != point1 or point2 == point1)
+
+    def testEqualityAlias(self):
+        """Test if == handles coordinate degeneracies correctly.
+        """
+        # Coordinate wrapping
+        self.assertEqual(SphPoint(360.0*degrees, -42.0*degrees),
+                         SphPoint(  0.0*degrees, -42.0*degrees))
+        self.assertEqual(SphPoint(-90.0*degrees, -42.0*degrees),
+                         SphPoint(270.0*degrees, -42.0*degrees))
+
+        # Polar degeneracy
+        self.assertEqual(SphPoint( 42.0*degrees,  90.0*degrees),
+                         SphPoint(270.0*degrees,  90.0*degrees))
+        self.assertEqual(SphPoint(-42.0*degrees, -90.0*degrees),
+                         SphPoint( 83.0*degrees, -90.0*degrees))
+
+        self.assertNotEqual(SphPoint(-42.0*degrees,  90.0*degrees),
+                            SphPoint( 83.0*degrees, -90.0*degrees))
+
+    def testInquality(self):
+        """Test if == and != give mutually consistent results.
+        """
+        for lon1, lat1 in self._dataset[0:20]:
+            point1 = SphPoint(lon1*degrees, lat1*degrees)
+            for lon2, lat2 in self._dataset[0:20]:
+                point2 = SphPoint(lon2*degrees, lat2*degrees)
+                self.assertTrue((point1 == point2) == (point2 == point1))
```

Comment:
This and the next two lines would be better expressed as using `assertEqual` and `assertNotEqual`, for example: `self.assertEqual((point1 == point2),  (point2 == point1))`. This gives more useful error messages.

Your use of `assertTrue(foo == bar)` is definitely appropriate when you are explicitly testing `operator==`, but otherwise `assertTrue` should be avoided if another assert is more expressive of what you are testing, both for readability and because the resulting error message is much more informative if the test fails.